main.go

```diff
@@ -18,45 +22,130 @@ var posts = []post{{
 	Title: "YOusdkfkd",
 }}
 
-func postsHandler(w http.ResponseWriter, r *http.Request) {
-	w.Header().Set("Content-Type", "application/json")
-
-	handlers := map[string]http.HandlerFunc{
-		"GET": func(w http.ResponseWriter, r *http.Request) {
-			resp, _ := json.Marshal(posts)
-			fmt.Fprintf(w, string(resp))
-		},
-		"POST": func(w http.ResponseWriter, r *http.Request) {
-			decoder := json.NewDecoder(r.Body)
-			var p post
-			if err := decoder.Decode(&p); err != nil {
-				panic(err)
-			}
-
-			id := len(posts) + 1
-			p.ID = id
-			posts = append(posts, p)
-			resp, _ := json.Marshal(p)
-			fmt.Fprintf(w, string(resp))
-		},
-		"DELETE": func(w http.ResponseWriter, r *http.Request) {
-			println(r.URL.Path)
-		},
-	}
-
-	handler, ok := handlers[r.Method]
-	if !ok {
-		fmt.Fprintf(w, "Method not supported")
+func find(id int) (int, error) {
+	for i, v := range posts {
+		if v.ID == id {
+			return i, nil
+		}
+	}
+
+	return 0, errors.New("Not found")
+}
```

Comment:
For a function like this, that can either return a success or one type of error, it's not uncommon to use the pattern of returning a boolean instead of an error:
```suggestion
func find(id int) (int, bool) {
	for i, v := range posts {
		if v.ID == id {
			return i, true
		}
	}

	return 0, false
}
```

This allows the user of this function to know whether it was found or not, and handle appropriately. For example, you could return "404" instead of returning an error.